File: PurchaseRequestApproval.Models/VendorContact.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace PurchaseRequestApproval.Models
{
    public class VendorContact
    {
        [Key]
        public int Id { get; set; }


        [Required]
        public string  Name { get; set; }

        [Required]
        public string Email { get; set; }

        public string Phone { get; set; }
        public string  Branch { get; set; }
        public string Remark { get; set; }
        [Required]
        public int VendorId { get; set; }
        [ForeignKey("VendorId")]
        public Vendor Vendor { get; set; }



    }
}

File: PurchaseRequestApproval/wwwroot/js/vendorcontact.js
var dataTable;
 
$(document).ready(function () {
    loadDataTable();
});


    function loadDataTable() {
        dataTable = $('#tblData').DataTable({
            "ajax": {
                "url": "/Admin/VendorContact/GetAll"
            },

            "columns": [

                { "data": "name", "width ": "15%" },
                { "data": "email", "width ": "15%" },
                { "data": "phone", "width ": "15%" },
                { "data": "branch", "width ": "15%" },
                { "data": "remark", "width ": "15%" },
                { "data": "vendor.vendorName", "width ": "15%" },




                {
                    "data": "id",
                    "render": function (data) {
                        return `
                            <div class="text-center">
                                <a href="/Admin/VendorContact/Upsert/${data}" class="btn btn-success text-white" style="cursor:pointer">
                                    <i class="fas fa-edit"></i> 
                                </a>
                                <a onclick=Delete("/Admin/VendorContact/Delete/${data}") class="btn btn-danger text-white" style="cursor:pointer">
                                    <i class="fas fa-trash-alt"></i> 
                                </a>
                            </div>
                           `;
                    }, "width": "10%"
                }
            ]
        });
}
function Delete(url) {
    swal({
        title: "Are you sure you want to Delete?",
        text: "You will not be able to restore the data!",
        icon: "warning",
        buttons: true,
        dangerMode: true
    }).then((willDelete) => {
        if (willDelete) {
            $.ajax({
                type: "DELETE",
                url: url,
                success: function (data) {
                    if (data.success) {
                        toastr.success(data.message);
                        dataTable.ajax.reload();
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            });
        }
    });
}
